app/controllers/ops_controller/diagnostics.rb

```diff
@@ -350,6 +347,47 @@ def diagnostics_server_list
 
   private ############################
 
+  def fetch_local_log(*params)
+    if MiqEnvironment::Command.supports_systemd?
+      fetch_journal_log(*params)
+    elsif !MiqEnvironment::Command.is_podified?
+      fetch_log_file(*params)
+    end
+  end
+
+  def fetch_journal_log(service_name = nil, max_count = 1_000, filter_params = {})
+    return unless MiqEnvironment::Command.supports_systemd?
+
+    filter_params[:syslog_identifier] = service_name if service_name.present?
+
+    require "systemd/journal"
+    Systemd::Journal.open do |journal|
+      journal.filter(filter_params)
+
+      entries = max_count.nil? ? journal.all : journal.take(max_count)
+      entries.map do |entry|
+        # This is the time in microseconds since the epoch UTC, formatted as a decimal string.
+        seconds_since_epoch = entry._source_realtime_timestamp.to_f / 1_000_000.0
+        timestamp = Time.strptime(seconds_since_epoch.to_s, "%s").strftime("%Y-%m-%dT%H:%M:%S.%6N ")
```

Comment:
Think this part can be simpler:

```suggestion
        timestamp = Time.at(seconds_since_epoch).strftime("%Y-%m-%dT%H:%M:%S.%6N ")
```

Reply:
:+1: done